build/Dockerfile

```diff
@@ -0,0 +1,101 @@
+# (c) Cartesi and individual authors (see AUTHORS)
+# SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+# syntax=docker.io/docker/dockerfile:1.4
+
+ARG RUST_VERSION=1.71.0
+ARG SERVER_MANAGER_VERSION=0.8.2
+ARG ROLLUPS_CONTRACTS_VERSION=1.0.2
+
+ARG BASE_PATH=/opt/cartesi
+ARG RUST_BUILD_PATH=${BASE_PATH}/src/rollups-node/offchain
+ARG DEPLOYMENT_PATH=${BASE_PATH}/share/deployments
+ARG RUNTIME_DIR=/var/opt/cartesi
+
+#
+# On-chain deployment files
+#
+FROM debian:bookworm-slim as deployment
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && apt-get install -y curl
+
+ARG DEPLOYMENT_PATH
+WORKDIR ${DEPLOYMENT_PATH}
+
+# Download deployment files from npm
+ARG ROLLUPS_CONTRACTS_VERSION
+RUN curl "https://registry.npmjs.org/@cartesi/rollups/-/rollups-${ROLLUPS_CONTRACTS_VERSION}.tgz" \
+        -o /tmp/rollups.tgz
+RUN tar zxf /tmp/rollups.tgz -C /tmp
+RUN cp /tmp/package/export/abi/* .
```

Comment:
This could be a `FROM scratch` image with just the contracts extracted. But it's debatable to have a container image that won't work with `docker run`, I have no problem with that.

---

It seems you installed `curl` to download the `.tgz`, we could save some time using `ADD`.

```suggestion
FROM debian:bookworm-slim as deployment

ARG DEPLOYMENT_PATH
WORKDIR ${DEPLOYMENT_PATH}

# Download deployment files from npm
ARG ROLLUPS_CONTRACTS_VERSION
ADD https://registry.npmjs.org/@cartesi/rollups/-/rollups-${ROLLUPS_CONTRACTS_VERSION}.tgz /tmp/rollups.tgz
RUN tar zxf /tmp/rollups.tgz -C /tmp
RUN cp /tmp/package/export/abi/* .
```

---

Is it possible to get a CHECKSUM from `npmjs`? It would be nice to have a verification for the integrity of the download.

Reply:
> This could be a FROM scratch image with just the contracts extracted. But it's debatable to have a container image that won't work with docker run, I have no problem with that.

Since this image is just a build step, we don't need to make it a scratch image.